src/adapters/mqtt/index.ts

```diff
@@ -20,105 +19,157 @@ class MqttAdapter extends Adapter {
-          qos: serverBinding && serverBinding.lastWill && serverBinding.lastWill.qos ? serverBinding.lastWill.qos : undefined,
-          payload: serverBinding && serverBinding.lastWill && serverBinding.lastWill.message ? serverBinding.lastWill.message : undefined,
-          retain: serverBinding && serverBinding.lastWill && serverBinding.lastWill.retain ? serverBinding.lastWill.retain : undefined,
-        },
-        keepalive: serverBinding && serverBinding.keepAlive,
-        username: userAndPasswordSecurityReq ? process.env.GLEE_USERNAME : undefined,
-        password: userAndPasswordSecurityReq ? process.env.GLEE_PASSWORD : undefined,
-        ca: X509SecurityReq ? certs : undefined,
-      } as any)
-
-      this.client.on('connect', () => {
-        if (!this.firstConnect) {
-          this.firstConnect = true
-          this.emit('connect', { name: this.name(), adapter: this, connection: this.client, channels: this.channelNames })
-        }
-
-        if (Array.isArray(subscribedChannels)) {
-          subscribedChannels.forEach((channel) => {
-            const operation = this.parsedAsyncAPI.channel(channel).publish()
-            const binding = operation.binding('mqtt')
-            this.client.subscribe(channel, {
-              qos: binding && binding.qos ? binding.qos : 0,
-            })
-          })
-        }
+      }
+    )
+    const userAndPasswordSecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'userPassword'
+    )
+    const X509SecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'X509'
+    )
+    const url = new URL(this.AsyncAPIServer.url())
+
+    this.client = mqtt.connect({
+      host: url.host,
+      port: url.port || (url.protocol === 'mqtt:' ? 1883 : 8883),
+      protocol: url.protocol.slice(0, url.protocol.length - 1),
+      clientId:
+        serverBinding &&
+        (serverBinding.clientId
+          ? serverBinding.clientId
+          : mqttOptions?.authentication?.clientId),
+      clean: serverBinding && serverBinding.cleanSession,
+      will: serverBinding &&
+        serverBinding.will && {
+        topic:
+          serverBinding &&
+            serverBinding.lastWill &&
+            serverBinding.lastWill.topic
+            ? serverBinding.lastWill.topic
+            : undefined,
+        qos:
+          serverBinding &&
+            serverBinding.lastWill &&
+            serverBinding.lastWill.qos
+            ? serverBinding.lastWill.qos
+            : undefined,
+        payload:
+          serverBinding &&
+            serverBinding.lastWill &&
+            serverBinding.lastWill.message
+            ? serverBinding.lastWill.message
+            : undefined,
+        retain:
+          serverBinding &&
+            serverBinding.lastWill &&
+            serverBinding.lastWill.retain
+            ? serverBinding.lastWill.retain
+            : undefined,
+      },
```

Comment:
```suggestion
      clientId: serverBinding?.clientId ?? mqttOptions?.authentication?.clientId,
      clean: serverBinding?.cleanSession,
      will: serverBinding?.will && {
        topic: serverBinding?.lastWill?.topic,
        qos: serverBinding?.lastWill?.qos,
        payload: serverBinding?.lastWill?.message,
        retain: serverBinding?.lastWill?.retain,
      },
```
@Souvikns I can't see why we are not using optional chaining here 🤔